### PR TITLE
ImageMagick with FOSS 2016b tool chain and FFmpeg support.

### PR DESCRIPTION
**Pull request by schiotz**
Easyconfig files for ImageMagick with the FOSS 2016b toolchain, and with FFmpeg as a dependency so ImageMagick can create movies.

### CODE REVIEW

**Comment by schiotz** [comment]
@boegel Your bot instructs me to talk to you if it acts stupid.  Does this qualify?
`AssertionError: No conflicts detected`
;-)

Of course looking into the detailed output from the bot reveals that there actually _is_ a conflict.  I will fix it and resubmit.

**Comment by boegel** [comment]
@schiotz There's a version conflicts on the LibTIFF and libjpeg-turbo dependencies.

You can check this locally using `eb --check-conflicts ImageMagick-7.0.5-10-foss-2016b.eb`

**Comment by boegel** [comment]
@schiotz No, this is a valid complaint. The tests assert that there are no conflicts, but conflicts were found, so the assertion fails.

**Comment by schiotz** [comment]
OK, I see the logic (but would probably have chosen the opposite text myself).

I have updated the dependencies.

File: easybuild/easyconfigs/i/ImageMagick/ImageMagick-7.0.5-10-foss-2016b.eb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+version = '7.0.5-10'
+
+homepage = 'http://www.imagemagick.org/'
+description = """ImageMagick is a software suite to create, edit, compose, or convert bitmap images"""
+
+toolchain = {'name': 'foss', 'version': '2016b'}
+
+sources = [SOURCE_TAR_GZ]
+source_urls = ['https://www.imagemagick.org/download/releases/']
+#checksums = ['473a1e7ae37548eb6fe9cf2b1fb78011']
```
Comment (boegel) [inline easybuild/easyconfigs/i/ImageMagick/ImageMagick-7.0.5-10-foss-2016b.eb]: @schiotz please provide right checksum?

File: easybuild/easyconfigs/i/ImageMagick/ImageMagick-7.0.5-10-foss-2016b.eb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+dependencies = [
+    ('bzip2', '1.0.6'),
+    #('X11', '20170314'),  # Conflicts with FFmpeg
```
Comment (boegel) [inline easybuild/easyconfigs/i/ImageMagick/ImageMagick-7.0.5-10-foss-2016b.eb]: please remove commented out line?

**Comment by schiotz** [comment]
@boegel Sorry about the checksum!  I did not know which kind of checksum it was, so I commented it out.  Then I forgot to add it back in.  It looks like it is MD5, it is back now and the easybuild builds with it, so I assume I guessed right :-)

**Comment by boegel** [comment]
lgtm

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in this PR)
node2423.golett.os - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/42f0816e0f5d95698cfa3a4c5120fb46 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in this PR)
node2056.delcatty.os - Linux centos linux 7.3.1611, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/ce5be7d8547703262257e112a25fe1ab for a full test report.

**Comment by schiotz** [comment]
@boegel 
Please hold the merging of this one.  It looks like there may be a problem with ImageMagick not picking up FFmpeg correctly.  It worked for me earlier, but now it looks problematic again.  It may be just be me, I will know later today or tomorrow (many meetings today), or it may be an actual problem with this PR.

Sorry about that!

**Comment by boegel** [comment]
@schiotz OK, just in time, I was about to hit merge on this one... ;-)

Let us know when you've figured things out.

**Comment by schiotz** [comment]
Weird.  ffmpeg dumps core on one of our machines, but not on another.  It looks like ffmpeg is loading way more libraries on the one where it dumps core, perhaps it has been build differently.  

Two things look suspicious to me, but I don't have time to get to the bottom of this today.  One is an encoding library which is strangely not included on the machine where it works.  The other is a NUMA library build with the Intel compiler which is loaded on the machine where it coredumps.  The latter looks way wrong to me.

I will let you know when I work it out.  It may just be our local setup.

**Comment by schiotz** [comment]
@boegel 
I think you can go ahead and merge it now.  It was a local weirdness causing trouble.  

I use our iomkl toolchain to compile some of my own software with Intel compilers and MPI for performance, and therefore load the toolchain in my .bashrc.  However, with that toolchain loaded the numactl library is replaced with an Intel compiled version.  This was causing ffmpeg to dump core.  If Intel compilers are not entering the picture, everything works as expected.

**Comment by boegel** [comment]
Going in, thanks @schiotz!

**Pull request merged**